### ISSUE TITLE
Partis

### ISSUE THREAD

**Issue by Pezhvuk** [matsengrp/sumrep#22]
Hi Branden, 

I'm going to leave this error here, to do with running the partis function. I might try and post an appropriate version of it on Partis github page, if I can extract more information from the partis out, and if we don't find a solution to it here. It would be great if you had any insight as to what is causing this issue, but, I am also trying changes in the source code, to illuminate on the problem a bit further. For instance, the warning message in the beginning is the result of using the `system(command, intern = TRUE)`. Couple of suspicious things would be the "NA" being passed as a directory, and "status 1" exit. The file is also attached.
[partis_test_file.txt](https://github.com/matsengrp/sumrep/files/2985344/partis_test_file.txt)

```
Warning message in system(command, intern = TRUE):
“running command '/bin/tcsh /d/as2/u/mp002/sumrep_14-03-2019/inst/run_partis.sh -p /d/as7/s/partis/bin/partis -a annotate -i /d/as2/u/mp002/sumrep_project/test/test_fasta/partis_RevC_test.fasta/NA -o /d/as2/u/mp002/sumrep_project/tests/partis/NA/partis_output.csv -n 4 -l BCR -h /d/as2/u/mp002/sumrep_project/tests/partis/NA/params -e v_gl_seq:v_qr_seqs:cdr3_seqs' had status 1”
Error in data.table::fread(., stringsAsFactors = TRUE): File '/d/as2/u/mp002/sumrep_project/tests/partis/NA/partis_output.csv' does not exist or is non-readable. getwd()=='/d/as2/u/mp002/sumrep_project/tests'
Traceback:

1. getPartisAnnotations(input_fileName, cleanup = TRUE, num_procs = NO_threads, 
 .     locus = "BCR", do_full_annotation = FALSE, output_path = output_file_directory)
2. callPartis(action = "annotate", input_filename = input_filename, 
 .     output_file = output_file, output_path = output_path, partis_path = partis_path, 
 .     num_procs = num_procs, locus = locus, extra_columns = extra_columns)   # at line 424-432 of file R/PartisFunctions.R
3. appendQuerySequencesToPartisAnnotationsFile(input_filename, annotation_filename)   # at line 65-68 of file R/PartisFunctions.R
4. annotation_filename %>% data.table::fread(stringsAsFactors = TRUE)   # at line 76-77 of file R/PartisFunctions.R
5. withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
6. eval(quote(`_fseq`(`_lhs`)), env, env)
7. eval(quote(`_fseq`(`_lhs`)), env, env)
8. `_fseq`(`_lhs`)
9. freduce(value, `_function_list`)
10. withVisible(function_list[[k]](value))
11. function_list[[k]](value)
12. data.table::fread(., stringsAsFactors = TRUE)
13. stop("File '", file, "' does not exist or is non-readable. getwd()=='", 
  .     getwd(), "'")
```

Cheers,
Peji.

**Comment by BrandenOlson** [comment]
Peji, 

This suggets to me that partis failed at some point and sumrep is trying to access a file that was never created. Indeed, it seems that a directory is not being specified correctly and thus converted to `NA`.

Would you mind posting the exact command you used to invoke `getPartisAnnotations`?

**Comment by Pezhvuk** [comment]
Hi Branden,

here is the simplest form of my code:

```
library(devtools)
library(ape)
devtools::load_all("/d/as2/u/mp002/sumrep_14-03-2019")

seq_path = "/d/as2/u/mp002/sumrep_project/test/test_fasta/S-FV_+14d.fasta"

partis_output <- getPartisAnnotations(seq_path, locus="ig")
```

And here is the error:

```
Loading sumrep
Warning message in system(command, intern = TRUE):
“running command '/bin/tcsh /d/as2/u/mp002/sumrep_14-03-2019/inst/run_partis.sh -p /d/as7/s/partis/bin/partis -a partition -i /d/as2/u/mp002/sumrep_project/test/test_fasta/S-FV_+14d.fasta -o _output/partis_output.csv -n 16 -l ig -h _output/params -e v_gl_seq:v_qr_seqs:cdr3_seqs' had status 1”
Error in data.table::fread(., stringsAsFactors = TRUE): File '_output/partis_output-cluster-annotations.csv' does not exist or is non-readable. getwd()=='/d/as2/u/mp002/sumrep_project/tests'
Traceback:

1. getPartisAnnotations(seq_path, locus = "ig")
2. getPartisPartitions(input_filename = input_filename, output_filename = output_filename, 
 .     partis_path = partis_path, num_procs = num_procs, locus = locus, 
 .     cleanup = FALSE, output_path = output_path, germline_dir = germline_dir, 
 .     extra_columns = extra_columns)   # at line 409-418 of file R/PartisFunctions.R
3. callPartis("partition", input_filename = input_filename, output_file = output_file, 
 .     output_path = output_path, partis_path = partis_path, num_procs = num_procs, 
 .     locus = locus, germline_dir = germline_dir, extra_columns = extra_columns)   # at line 469-478 of file R/PartisFunctions.R
4. appendQuerySequencesToPartisAnnotationsFile(input_filename, annotation_filename)   # at line 65-68 of file R/PartisFunctions.R
5. annotation_filename %>% data.table::fread(stringsAsFactors = TRUE)   # at line 76-77 of file R/PartisFunctions.R
6. withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
7. eval(quote(`_fseq`(`_lhs`)), env, env)
8. eval(quote(`_fseq`(`_lhs`)), env, env)
9. `_fseq`(`_lhs`)
10. freduce(value, `_function_list`)
11. withVisible(function_list[[k]](value))
12. function_list[[k]](value)
13. data.table::fread(., stringsAsFactors = TRUE)
14. stop("File '", file, "' does not exist or is non-readable. getwd()=='", 
  .     getwd(), "'")
```

Cheers,
Peji.

**Comment by Pezhvuk** [comment]
btw, I ran partis on the same machines last year, and I didn't have this issue as far as I can remember.

**Comment by Pezhvuk** [comment]
solved in #23
